aws_lambda_powertools/utilities/data_classes/kinesis_firehose_event.py

```diff
@@ -0,0 +1,113 @@
+import base64
+import json
+from typing import Iterator, Optional
+
+from aws_lambda_powertools.utilities.data_classes.common import DictWrapper
+
+
+class KinesisFirehoseRecordMetadata(DictWrapper):
+    @property
+    def _metadata(self) -> dict:
+        """Optional: metadata associated with this record; present only when Kinesis Stream is source"""
+        return self["kinesisRecordMetadata"]  # could raise KeyError
+
+    @property
+    def shard_id(self) -> Optional[str]:
+        """Kinesis stream shard ID; present only when Kinesis Stream is source"""
+        return self._metadata.get("shardId")
+
+    @property
+    def partition_key(self) -> Optional[str]:
+        """Kinesis stream partition key; present only when Kinesis Stream is source"""
+        return self._metadata.get("partitionKey")
+
+    @property
+    def approximate_arrival_timestamp(self) -> Optional[str]:
+        """Kinesis stream approximate arrival ISO timestamp; present only when Kinesis Stream is source"""
+        return self._metadata.get("approximateArrivalTimestamp")
+
+    @property
+    def sequence_number(self) -> Optional[str]:
+        """Kinesis stream sequence number; present only when Kinesis Stream is source"""
+        return self._metadata.get("sequenceNumber")
+
+    @property
+    def subsequence_number(self) -> Optional[str]:
+        """Kinesis stream sub-sequence number; present only when Kinesis Stream is source
+
+        Note: this will only be present for Kinesis streams using record aggregation
+        """
+        return self._metadata.get("subsequenceNumber")
+
+
+class KinesisFirehoseRecord(DictWrapper):
+    @property
+    def approximate_arrival_timestamp(self) -> float:
+        """The approximate time that the record was inserted into the delivery stream"""
+        return float(self["approximateArrivalTimestamp"])
+
+    @property
+    def record_id(self) -> str:
+        """Record ID; uniquely identifies this record within the current batch"""
+        return self["recordId"]
+
+    @property
+    def data(self) -> str:
+        """The data blob, base64-encoded"""
+        return self["data"]
+
+    @property
+    def metadata(self) -> KinesisFirehoseRecordMetadata:
+        """Optional: metadata associated with this record; present only when Kinesis Stream is source"""
+        return KinesisFirehoseRecordMetadata(self._data)
+
+    @property
+    def data_as_bytes(self) -> bytes:
+        """Decoded base64-encoded data as bytes"""
+        return base64.b64decode(self.data)
+
+    @property
+    def data_as_text(self) -> str:
+        """Decoded base64-encoded data as text"""
+        return self.data_as_bytes.decode("utf-8")
+
+    @property
+    def data_as_json(self) -> dict:
+        """Decoded base64-encoded data loaded to json"""
+        if self._json_data is None:
+            self._json_data = json.loads(self.data_as_text)
+        return self._json_data
```

Comment:
I think we need to review this piece of code. 
This is not your fault, we have this method/property as default in all classes and this can be a problem for records that allow non-json data like `firehose put` is and maybe others.. Check this out

![image](https://user-images.githubusercontent.com/4295173/195993705-c2d99671-20a7-4632-918e-ce9ebb94e859.png)
![image](https://user-images.githubusercontent.com/4295173/195993724-ffee87bc-7890-45c9-92d0-d4c2187f127c.png)
![image](https://user-images.githubusercontent.com/4295173/195993739-37d701f7-9eca-4063-8935-072673a3fa28.png)

We'll discuss this in our daily sync on Monday and I back here with updates.

Reply:
EAFP